torchgeo/datasets/forestdamage.py

```diff
@@ -0,0 +1,327 @@
+        """Return the number of data points in the dataset.
+
+        Returns:
+            length of the dataset
+        """
+        return len(self.files)
+
+    def _load_files(self, root: str) -> List[Dict[str, str]]:
+        """Return the paths of the files in the dataset.
+
+        Args:
+            root: root dir of dataset
+
+        Returns:
+            list of dicts containing paths for each pair of image, annotation
+        """
+        images = sorted(
+            glob.glob(os.path.join(root, self.data_dir, "**", "Images", "*.JPG"))
+        )
+        annotations = sorted(
+            glob.glob(os.path.join(root, self.data_dir, "**", "Annotations", "*.xml"))
+        )
+
+        files = [
+            dict(image=image, annotation=annotation)
+            for image, annotation in zip(images, annotations)
+        ]
+
+        return files
+
+    def _load_image(self, path: str) -> Tensor:
+        """Load a single image.
+
+        Args:
+            path: path to the image
+
+        Returns:
+            the image
+        """
+        with Image.open(path) as img:
+            array: "np.typing.NDArray[np.int_]" = np.array(img.convert("RGB"))
+            tensor: Tensor = torch.from_numpy(array)
+            # Convert from HxWxC to CxHxW
+            tensor = tensor.permute((2, 0, 1))
+            return tensor
+
+    def _load_target(
+        self, bboxes: List[List[int]], labels_list: List[str]
+    ) -> Tuple[Tensor, Tensor]:
+        """Load the target mask for a single image.
+
+        Args:
+            bboxes: list of bbox coordinats [xmin, ymin, xmax, ymax]
+            labels_list: list of class labels
+
+        Returns:
+            the target bounding boxes and labels
+        """
+        labels = torch.tensor([self.class_to_idx[label] for label in labels_list])
+        boxes = torch.tensor(bboxes).to(torch.float)
+        return boxes, labels
+
+    def _verify(self) -> None:
+        """Checks the integrity of the dataset structure.
+
+        Returns:
+            True if the dataset directories are found, else False
+        """
+        filepath = os.path.join(self.root, self.data_dir)
+        if os.path.isdir(filepath):
+            return
+
+        filepath = os.path.join(self.root, self.data_dir + ".zip")
+        if os.path.isfile(filepath):
+            if self.checksum and not check_integrity(filepath, self.md5):
+                raise RuntimeError("Dataset found, but corrupted.")
+            extract_archive(filepath)
+            return
+
+        # Check if the user requested to download the dataset
+        if not self.download:
+            raise RuntimeError(
+                "Dataset not found in `root` directory, either specify a different"
+                + " `root` directory or manually download "
+                + "the dataset to this directory."
+            )
+
+        # else download the dataset
+        self._download()
+
+    def _download(self) -> None:
```

Comment:
That is what I intended with the step above line 225-230, where if the downloaded zip file is found, it will only be extracted. Or would you prefer that in the case no zip file is found, the download and extract also happen in two different steps?

Reply:
Ah, I guess this is fine then. I usually have separate `_download` and `_extract` functions but in this case it's just a single file.